Styles: Load Bootstrap Once

ewlarson opened the pull request:
Currently, both Blacklight and GeoBlacklight import Bootstrap, which causes that large library to appear twice in the final application.css file.

This PR does a few good things:

1. Reorganizes GBL stylesheets to be better aligned with BL
1. Extracts the asset generator steps into its own generator file
1. Overwrites the default Rails application.css file and and the default BL blacklight.scss file to help us avoid bootstrap duplication
1. Creates a simple flow for making local customizations, and importing bootstrap, blacklight, and geoblacklight style rules.

Fixes #854

mejackreed commented:
We should coordinate the merge of this PR with a next major version release as it seems like there will be some breaking changes introduced here.

ewlarson commented:
- [ ] Update [style customization tutorial](https://geoblacklight.org/tutorial/2015/02/09/customize-your-application.html#changing-the-style-of-your-application) upon release

ewlarson commented:
@eliotjordan / I removed the prior cache bust commits and rebased. Had to commit another cache bust to get the .internal_test_app to build with the new asset generator step. Should be all set to go now, except the coverage step in the CI matrix is failing -- says solr_wrapper gem can't be found:
https://app.circleci.com/pipelines/github/geoblacklight/geoblacklight/466/workflows/3ae0bdb7-cf13-4c1d-a565-6cd0cc622193/jobs/3907

Anyway, this is the last hurdle to merging this one. If you have any suggestions for the coverage step, please let me know!
